mixer/adapter/kubernetesenv/kubernetesenv.go

```diff
@@ -138,30 +149,52 @@ func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler,
 	if !exists {
 		path = paramsProto.KubeconfigPath
 	}
-	clientset, err := b.newClientFn(path, env)
+
+	config, err := b.newConfig(path, env)
 	if err != nil {
-		return nil, err
-	}
-	controller := newCacheController(clientset, refresh)
-	env.ScheduleDaemon(func() { controller.Run(stopChan) })
-	// ensure that any request is only handled after
-	// a sync has occurred
-	env.Logger().Infof("Waiting for kubernetes cache sync...")
-	if success := cache.WaitForCacheSync(stopChan, controller.HasSynced); !success {
-		stopChan <- struct{}{}
-		return nil, errors.New("cache sync failure")
-	}
-	env.Logger().Infof("Cache sync successful.")
+		return nil, fmt.Errorf("could not build config: %v", err)
+	}
+
+	var controller cacheController
+	key := configKey(config)
```

Comment:
I think most of the cases the controller (and its config) can be keyed by the `path` rather than the internal key object. Sure two different config paths can point to the same API server, but that's quite rare -- and using path as the key would simplify the code a lot. wdyt?

Reply:
sgtm. updated to use `path` and switched to a simpler map for the controllers. ptal.